src/audio/google/google_rtc_audio_processing_mock.c

```diff
@@ -187,18 +214,37 @@ int GoogleRtcAudioProcessingProcessCapture_int16(GoogleRtcAudioProcessingState *
 						 const int16_t *const src,
 						 int16_t *const dest)
 {
+	if (!state || !src || !dest || !state->aec_reference)
+		return -1; // Return an error code if any of the pointers are null
```

Comment:
Lets have proper errno codes for all negative error numbers. Most of these are passed up the stack and could result in confusion if we are passing wrong codes.

Reply:
Done - please have a look

Reply:
Here and some lines below still `-1`  in use. @thesofproject/google 's comments would be useful for error codes from mockup. If the real library uses similar, returning them would be better if the error code macros are possible to share in this public open-source version.

Reply:
The internal library does not have error code. 0 means success any negative value mean error.